src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -248,17 +263,44 @@ private static boolean shouldBeOnSameLine(RightCurlyOption bracePolicy, Details
      * @return true if a right curly should be alone on a line.
      */
     private static boolean shouldBeAloneOnLine(RightCurlyOption bracePolicy, Details details) {
+        final int tokenLambda = details.lcurly.getParent().getType();
         return bracePolicy == RightCurlyOption.ALONE
+                && tokenLambda != TokenTypes.LAMBDA
                 && !isAloneOnLine(details)
                 && !isEmptyBody(details.lcurly)
-                || bracePolicy == RightCurlyOption.ALONE_OR_SINGLELINE
+                || shouldBeAloneInLambda(bracePolicy, details)
+                || shouldBeAloneOrSingleLine(bracePolicy, details);
+    }
+
+    /**
+     * Checks that a right curly should be alone on a line or allows single-line format of block.
+     * @param bracePolicy option for placing the right curly brace
+     * @param details Details for validation
+     * @return true if a right curly should be alone on a line or single-line format of block.
+     */
+    private static boolean shouldBeAloneOrSingleLine(RightCurlyOption bracePolicy,
+                                                     Details details) {
+        return bracePolicy == RightCurlyOption.ALONE_OR_SINGLELINE
                 && !isAloneOnLine(details)
                 && !isSingleLineBlock(details)
                 && !isAnonInnerClassInit(details.lcurly)
                 && !isEmptyBody(details.lcurly);
     }
 
     /**
+     * Checks that a right curly should be alone on a line in the token is LAMBDA.
+     * @param bracePolicy option for placing the right curly brace
+     * @param details Details for validation
+     * @return true if after right curly should be on line in the token is LAMBDA.
+     */
+    private static boolean shouldBeAloneInLambda(RightCurlyOption bracePolicy, Details details) {
+        final int tokenLambda = details.lcurly.getParent().getType();
+        return bracePolicy == RightCurlyOption.ALONE
+                && tokenLambda == TokenTypes.LAMBDA
```

Comment:
Now the upper methods further for them is validate(). And bring these checks in it we will get "cyclomatic complexity".  Can as I wrote above create a new common further method for them.
Or so, to do 

private static boolean **isValidAloneOnLine**(RightCurlyOption bracePolicy,
                                                      Details details) {
        final int tokenLambda = details.lcurly.getParent().getType();
        return tokenLambda != TokenTypes.LAMBDA
                && shouldBeAloneOnLine(bracePolicy, details);
                || tokenLambda == TokenTypes.LAMBDA
                && shouldBeAloneOther(bracePolicy, details);
    }

private static boolean **isValidAloneOnLineOther**(RightCurlyOption bracePolicy,
                                                      Details details) {
        final int tokenLambda = details.lcurly.getParent().getType();
        return  tokenLambda == TokenTypes.LAMBDA
                && shouldBeAloneOther(bracePolicy, details);
    }

and in validate()

else if (isValidAloneOnLine(bracePolicy, details)
          && isValidAloneOnLineOther(bracePolicy, details)) {
            violation = MSG_KEY_LINE_ALONE;
        }